search/search_engine.cpp

```diff
@@ -151,11 +152,37 @@ void Engine::SetViewportAsync(m2::RectD const & viewport)
   m_query->SetViewport(r, true);
 }
 
-void Engine::EmitResults(SearchParams const & params, Results & res)
+void Engine::EmitResults(SearchParams const & params, m2::RectD const & viewport, Results & res)
 {
-  // Basic test of our statistics engine.
-  alohalytics::LogEvent("searchEmitResults",
-                        alohalytics::TStringMap({{params.m_query, strings::to_string(res.GetCount())}}));
+  size_t const kMaxNumResultsToSend = 10;
+
+  size_t numResultsToSend = min(kMaxNumResultsToSend, res.GetCount());
+  string resultString = strings::to_string(numResultsToSend);
+  for (size_t i = 0; i < numResultsToSend; ++i)
+    resultString.append("\t" + DebugPrint(static_cast<search::Result const &>(res.GetResult(i))));
+
+  double lat = -1;
+  double lon = -1;
+  if (params.IsValidPosition())
+  {
+    lat = params.m_lat;
+    lon = params.m_lon;
+  }
+
+  alohalytics::TStringMap stats = {
+      {"lat", strings::to_string(lat)},
+      {"lon", strings::to_string(lon)},
+      {"viewportMinX", strings::to_string(viewport.minX())},
```

Comment:
Am I right that viewport coords are in Mercator coordinate system? If so, may be it's worth to unify types of user position and viewport corners, i.e. to use either Mercator either Lat/Lon?